aps/utils/roxar/grid_model.py

```diff
@@ -125,15 +135,33 @@ def average_of_property_inside_zone_region(
     Returns a dictionary with parameter name as key and average as value.
     """
     cell_index_defined = find_defined_cells(
-        zone_values, zone_number, region_values, region_number, debug_level=Debug.OFF
+        zone_values, zone_number, region_values, region_number, debug_level=debug_level
     )
-    return {
-        name: calc_average(
-            cell_index_defined,
-            values=getContinuous3DParameterValues(grid_model, name, realization_number),
-        )
-        for name in parameter_names
-    }
+
+    if debug_level >= Debug.VERY_VERBOSE:
+        check_for_nan(cell_index_defined, 'cell_index_defined')
+
+        average_dict = {}
+        for name in parameter_names:
+            values = getContinuous3DParameterValues(
+                grid_model, name, realization_number
+            )
+            check_for_nan(values, name)
+            average_dict[name] = calc_average(
+                cell_index_defined, values=values, debug_level=debug_level
+            )
+        return average_dict
+    else:
+        return {
+            name: calc_average(
+                cell_index_defined,
+                values=getContinuous3DParameterValues(
+                    grid_model, name, realization_number
+                ),
+                debug_level=debug_level,
+            )
+            for name in parameter_names
+        }
```

Comment:
Not sure if I love that changing the `debug_level` also changes what code is run and what is returned from a function. In my mind, the `debug_level` should only add console output.

Reply:
But this too changes the flow a bit, since adding the `check_for_nan` actually adds the possibility for an error to be raised.

Reply:
I will test this on a real case and check if there are any notable increase in run-time or GUI response before merging. If, not I can include the check to always be run instead of conditional running depending on debug level.